sdk/identity/identity/src/credentials/managedIdentityCredential/arcMsi.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AccessToken, GetTokenOptions, RequestPrepareOptions } from "@azure/core-http";
+import { MSI } from "./models";
+import { credentialLogger } from "../../util/logging";
+import { IdentityClient } from "../../client/identityClient";
+import { msiGenericGetToken } from "./utils";
+import { azureArcAPIVersion } from "./constants";
+import { AuthenticationError } from "../../client/errors";
+import { readFile } from "fs";
+
+const logger = credentialLogger("ManagedIdentityCredential - ArcMSI");
+const defaultArcMsiEndpoint = "http://localhost:40342/metadata/identity/oauth2/token";
+
+// Azure Arc MSI doesn't have a special expiresIn parser.
+const expiresInParser = undefined;
+
+function prepareRequestOptions(resource?: string, clientId?: string): RequestPrepareOptions {
+  const queryParameters: any = {
+    resource,
+    "api-version": azureArcAPIVersion
+  };
+
+  if (clientId) {
+    queryParameters.client_id = clientId;
+  }
+
+  return {
+    url: process.env.IMDS_ENDPOINT || defaultArcMsiEndpoint,
```

Comment:
I wonder if we should do this, or instead throw an error if the environment variable isn't available. The endpoint won't change, but the environment variable being absent might suggest that the user's environment is no longer configured correctly. The request would probably fail anyway if that's the case, but I'd be curious to hear @catalinaperalta's thoughts on how to handle this case

Reply:
I'm ok on whatever Catalina says 🌞 thank you for bringing this up, McCoy!

Reply:
I think this comment can be resolved now? I didn't see this in the current implementation. The way it is now with only IDENTITY_ENDPOINT is correct.